src/main/java/graphql/nadel/Nadel.java

```diff
@@ -50,7 +53,7 @@
 
     private final Logger log = LoggerFactory.getLogger(Nadel.class);
     private final StitchingDsl stitchingDsl;
-    private final ServiceDataFactory serviceDataFactory;
+    private final ServiceExecutionFactory serviceExecutionFactory;
```

Comment:
I had to change the ServiceDataFactory signature - I took the opportunity to rename interface.  This is a breaking change either way